### PR TITLE
Code for Interpolation Search algorithm in C++ and Python Added

### PR DESCRIPTION
**Pull request by sharur7**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

The Interpolation Search Algorithm:
Interpolation search is an algorithm for searching for a key in an array that has been ordered by numerical values assigned to the keys.
It is an improved variant of binary search.
It has not yet been added to this repository.
I have added code for this in C++ and Python.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#451

### CODE REVIEW

**Comment by sharur7** [comment]
@plazzy99 @Kajol-Kumari @Abhijit2505 Please review this PR. Thank You!

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@sharur7 please format the code properly since there is a lot of indentations try to use code formatters. Also, avoid using `lo` and `hi` rather try to use a meaningful variable like `low` and `high`. Add sample input-output and time and space complexity at the end of the code.

File: C-Plus-Plus/search/Interpolation_Search.cpp [hunk 1]
```diff
@@ -1,78 +1,102 @@
...
+  if (index != -1)
+    cout << "Element found at index " << index; //Returning the index pf the element
+  else
+    cout << "Element not found."; //Returning -1 as element has not found 
+
+  return 0;
+}
+
+
+//SAMPLE INPUT:
```
Comment (iamrajiv) [inline C-Plus-Plus/search/Interpolation_Search.cpp]: @sharur7 update this to multi-line comment something like this
```
/*
...
*/
```

**Review by iamrajiv — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/search/Interpolation_Search.cpp (hunk 1 above)
Comment (sharur7) [reply C-Plus-Plus/search/Interpolation_Search.cpp]: Yes, I did the changes in both C++ and Python files.

**Review by siddharth25pandey — APPROVED** [review]
@sharur7 Looks fine to me.
Able to merge.
Thanks!

File: Python/search/Interpolation_Search.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
+# INTERPOLATION SEARCH
+
+# If the searching element is present in array, then returns index of it, else returns -1
+
+
+def interpolationSearch(arr, n, x):
```
Comment (SKAUL05) [inline Python/search/Interpolation_Search.py]: Use proper naming convention as per PEP8

File: Python/search/Interpolation_Search.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+# Creating array of the size n
+# Size would be taken into the variable n from user
+
+arr = []
+n = int(input('Enter the size of array : '))
+
+# Entering the elements in the array
+
+print 'Enter elements :'
+for i in range(0, n):
```
Comment (SKAUL05) [inline Python/search/Interpolation_Search.py]: Use meaningful variables instead of alphabets

File: Python/search/Interpolation_Search.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+            high = pos - 1
+
+    return -1
+
+
+# Starting with the main code
+# Creating array of the size n
+# Size would be taken into the variable n from user
+
+arr = []
```
Comment (SKAUL05) [inline Python/search/Interpolation_Search.py]: Include this whole part under __name__ == "__main__"

**Review by SKAUL05 — CHANGES_REQUESTED** [review]

File: Python/search/Interpolation_Search.py (hunk 2 above)
Comment (sharur7) [reply Python/search/Interpolation_Search.py]: Done

File: Python/search/Interpolation_Search.py (hunk 3 above)
Comment (sharur7) [reply Python/search/Interpolation_Search.py]: Done

File: Python/search/Interpolation_Search.py (hunk 4 above)
Comment (sharur7) [reply Python/search/Interpolation_Search.py]: I defined main() function to the specified code.
Please check the commit "Updated Interpolation_Search.cpp". I made a mistake in commit the message it should end with .py extension.

File: Python/search/Interpolation_Search.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+# Creating array of the size n
+# Size would be taken into the variable n from user
+def main():
+    arr = []
+    size = int(input('Enter the size of array : '))
+
+    # Entering the elements in the array
+
+    print ('Enter elements :')
+    for i in range(0, size):
```
Comment (SKAUL05) [inline Python/search/Interpolation_Search.py]:
```suggestion
    for index in range(0, size):
```

**Review by SKAUL05 — CHANGES_REQUESTED** [review]

File: Python/search/Interpolation_Search.py (hunk 5 above)
Comment (sharur7) [reply Python/search/Interpolation_Search.py]: Done

**Review by SKAUL05 — APPROVED** [review]

**Review by iamrajiv — APPROVED** [review]

**Comment by iamrajiv** [comment]
@all-contributors please add @sharur7 for code

**Pull request merged**